cms/models/pagemodel.py

```diff
@@ -833,25 +835,21 @@ def mark_as_pending(self, language):
                 published=False
             )
 
-        draft = self.get_draft_object()
-
-        if draft and draft.is_published(language) and draft.get_publisher_state(
-                language) == PUBLISHER_STATE_DEFAULT:
+        if self.is_published(language) and self.get_publisher_state(language) == PUBLISHER_STATE_DEFAULT:
             # Only change the state if the draft page is published
             # and it's state is the default (0)
-            draft.set_publisher_state(language, state=PUBLISHER_STATE_PENDING)
+            self.set_publisher_state(language, state=PUBLISHER_STATE_PENDING)
 
     def mark_descendants_pending(self, language):
         assert self.publisher_is_draft
```

Comment:
There's a lot of asserts like this for internal page methods.
I decided to leave them there because they prevent anyone using these internals from corrupting the db and also help when reading through the code to understand that these are only meant to work with draft (or public) pages.